server/src/main/java/org/opensearch/common/cache/stats/CacheStatsSnapshot.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.stats;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * An immutable snapshot of CacheStats.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class CacheStatsSnapshot implements Writeable { // TODO: Make this extend ToXContent (in API PR)
```

Comment:
Should we rename this class also to ImmutableCacheStats?

Reply:
Yes - my bad I missed this one